### ISSUE TITLE
Add Test Cases for `UploadToCloudStorage`

### ISSUE THREAD

**Issue by uzaxirr** [ZeStream/zestream-server#53]
Add Test cases for the function `UploadToCloudStorage` and all possible functions in `utils/upload.go`

The **cloud provider services needs to be mocked** over here (Eg: AWS, GCP and Azure)

**Comment by uzaxirr** [comment]
Hey @Shofiya2003 
Would you like to work on this?

**Comment by Shofiya2003** [comment]
Sure

**Comment by uzaxirr** [comment]
> Sure

Thanksss, Assigning it to you right away!!

**Comment by Shofiya2003** [comment]
@uzaxirr @abhi-shek-raj apologies for the delay 
will work on it this week

**Comment by uzaxirr** [comment]
> @uzaxirr @abhi-shek-raj apologies for the delay will work on it this week

Not a problem at all!

**Comment by Shofiya2003** [comment]
I am using the documentation of every cloud service provider to mock the service in the unit test. 
Wrapping my heads around documentation.
Do you have any suggestions for writing tests to mock the cloud services using another go package? 
Also, i will have to test the function which extracts the files out of the folders Right?
@uzaxirr

**Comment by Shofiya2003** [comment]
I started with aws and am using mock client and created logs for the process to test the service. 
Should I create a separate test for every cloud service or try to include the tests in the `UploadToCloudStorage` function?
@uzaxirr @abhishekraj272

**Comment by uzaxirr** [comment]
> Should I create a separate test for every cloud service or try to include the tests in the UploadToCloudStorage function?

Yes, in this way it will be better to segregate the services and have individual customisation for each if or whenever required


> Do you have any suggestions for writing tests to mock the cloud services using another go package?

Checkout [this](https://github.com/ZeStream/zestream-server/issues/22#issuecomment-1368421069) comment, see if you can do something similar.

> I will have to test the function which extracts the files out of the folders Right?

I didn't got this one, can you mention the function name please.

**Comment by Shofiya2003** [comment]
hey! @uzaxirr 
I am trying to create mock for aws upload but [mock interface](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/s3iface/#S3API) does not have the `Upload` function we are using. It has the `PutObject` and `MultipartUpload` . After reading it seems `Upload` uses either `PutObject` or `MultipartUpload` depending on the size of the upload.
I used [pushback request handler](https://docs.aws.amazon.com/sdk-for-go/api/aws/request/#HandlerList.PushBack) to generate the log of upload process using [unit test session of AWS] (https://pkg.go.dev/github.com/aws/aws-sdk-go/awstesting/unit) and then compared this log to the expected log.
I am looking for ways to mock the upload service using https://docs.aws.amazon.com/sdk-for-go/api/service/s3/s3iface/#S3API

**Comment by Shofiya2003** [comment]
> I didn't got this one, can you mention the function name please.

The `WalkFunc` which extracts the files from the folder path

**Comment by Shofiya2003** [comment]
@uzaxirr  @abhishekraj272  can we divide this into three issues each for the three cloud storage platform. I completed implementing the tests for aws upload.

**Comment by abskrj** [comment]
Yes works, just create the tickets, give some description and assign those
on which you are working
